Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -203,47 +203,46 @@ function xASL_wrp_ResampleASL(x)
 
 
 %% ------------------------------------------------------------------------------------------
-% 8. Pair-wise subtraction & saving PWI & PWI4D in both spaces
+% 8. Pair-wise control-label subtraction
+% for both native & standard space
+
+% Nomenclature:
+% ASL4D = original timeseries
+% PWI4D = subtracted timeseries -> for quantification purposes
+% PWI3D = averaged per TE—PLD—LabDur combination -> for QC purposes
+% PWI   = single perfusion-weighted volume -> for registration purposes
+
+% 
+
 % Load ASL time series (after being pre-processed)
 StringSpaceIs = {'native' 'standard'};
-PathASL = {x.P.Path_rdespiked_ASL4D x.P.Path_rtemp_despiked_ASL4D};
+PathASL4D = {x.P.Path_rdespiked_ASL4D x.P.Path_rtemp_despiked_ASL4D};
 PathPWI = {x.P.Path_PWI x.P.Pop_Path_PWI};
+PathPWI3D = {x.P.Path_PWI3D x.P.Pop_Path_PWI3D};
 PathPWI4D = {x.P.Path_PWI4D x.P.Pop_Path_PWI4D};
 
 for iSpace=1:2
     fprintf('%s\n', ['Saving in ' StringSpaceIs{iSpace} ' space:']);
     
-    ASL_im = xASL_io_Nifti2Im(PathASL{iSpace}); % Load time-series nifti
+    ASL_im = xASL_io_Nifti2Im(PathASL4D{iSpace}); % Load time-series nifti
     dim4 = size(ASL_im, 4);
 
-    if dim4==1
-        % Apparently, the subtraction was already done on the scanner/reconstruction
-        fprintf('%s\n', PathPWI{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, ASL_im, 32, false);
-	elseif x.modules.asl.bContainsDeltaM
-		% The original ASL4D volume contains subtraction images and not control/label images
-		
-		% For multi-PLD, save both PWI4D and PWI
-		if x.modules.asl.bMultiPLD
-			% Save PWI4D
-			fprintf('%s\n', PathPWI4D{iSpace});
-			xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, ASL_im, 32, false);
-		end
-		
-		% For both single- and multi-PLD, create single PWI for further steps in ASL module
-		PWIsingle = xASL_stat_MeanNan(ASL_im,4); % Average across PLDs
+    
+    % =====================================================================
+    % A) Check subtraction
+    if numel(size(ASL_im))>4
+        error('Unknown multi-dimensionality of ASL4D.nii');

```

Comment:
I wouldn't understand this message. I would write this more obviously. Something like:
All multi-dimensional ASL images are saved as 4D, but more dimensions than 4 are detected here.

Reply:
Done